src/types/utils/EmptyObject.ts

```diff
@@ -0,0 +1,19 @@
+import Falsy from './Falsy';
+
+type EmptyObject = Record<string, never>;
+
+// eslint-disable-next-line rulesdir/no-negated-variables
+function isNotEmptyObject<T extends Record<string, unknown> | Falsy>(arg: T | EmptyObject): arg is T {
+    return Object.keys(arg ?? {}).length > 0;
+}
```

Comment:
```suggestion
function isNotEmptyObject<T extends Record<string, unknown> | Falsy>(arg: T): arg is T {
    return Object.keys(arg ?? {}).length > 0;
}
```

`EmptyObject` is unnecessary here.

Reply:
updated